src/main/clojure/jarl/util.clj

```diff
@@ -1,17 +1,7 @@
 (ns jarl.util)
 
 (defn map-by-index [array]
-  (loop [array array
-         map {}
-         i 0]
-    (if (empty? array)
-      map
-      (recur (next array) (assoc map i (first array)) (inc i)))))
+  (zipmap (range (count array)) array))
 
 (defn indexed-map-to-array [map]
-  (let [max-index (key (apply max-key key map))]
-    (loop [i 0
-           array []]
-      (if (> i max-index)
-        array
-        (recur (inc i) (conj array (get map i)))))))
+  (vec (vals (into (sorted-map) map))))
```

Comment:
Won't this drop missing entries in the key set? E.g. `{0 "a" 3 "d" 4 "e"}` -> `["a", "d", "e"]` instead of the expected `["a" null null "d" "e"]`? 
This function is used to feed built-in functions their arguments, and it's important that the key in the incoming map corresponds to that value's index in the outgoing list.

Of course, the `null` values should eventually be that `Undefined` type we discussed out-of-band earlier.

Reply:
Oh, interesting! Will check.

Reply:
Yep, I've reverted that and added a docstring to explain.